src/components/profile/profilemenu/ProfileMenu.js

```diff
@@ -0,0 +1,262 @@
+import Text from "@/components/base/text/Text";
+import Title from "@/components/base/title";
+import Link from "@/components/base/link";
+import Icon from "@/components/base/icon/Icon";
+import styles from "@/components/profile/profilemenu/ProfileMenu.module.css";
+import Translate from "@/components/base/translate/Translate";
+import { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import classNames from "classnames/bind";
+import { encodeString } from "@/lib/utils";
+import { getElementById } from "@/lib/utils";
+import useUser from "@/components/hooks/useUser";
+
+/**
+ * This component shows a profile menu on the left handside of the profile page.
+ * It contains two types of links:
+ * simple links such as "Mine bilioteker"
+ * and links with subcategories such as "Lån og reserveringer" with subcateogries "Lån", "Reserveringer", "Mellemværende".
+ * @returns {JSX.Element}
+ */
+
+/**
+ * Simple menu link without subcategories.
+ *
+ * @returns {JSX.Element}
+ */
+function MenuLink({ label, href }) {
+  const router = useRouter();
+  const [isActive, setIsActive] = useState(router.asPath.includes(href));
+
+  useEffect(() => {
+    setIsActive(router.asPath.includes(href));
+  }, [router.asPath]);
+
+  const type = isActive ? "title4" : "title5";
+  const activeClass = isActive ? styles.active : "";
+
+  return (
+    <div className={`${styles.link} ${activeClass}`}>
+      <Link href={href} dataCy="menu-fixed-links">
+        <Title type={type}>{Translate({ context: "profile", label })}</Title>
+      </Link>
+      <Icon src="arrowrightblue.svg" size={1} />
+    </div>
+  );
+}
+
+/**
+ * One group of subcategories in the menu.
+ * f. ex. "Lån", "Reserveringer", "Mellemværende" under "Lån og reserveringer"
+ * @param menuItems
+ * @param href
+ * @param groupName
+ * @param activeIndex
+ * @param setActiveIndex
+ * @return {JSX.Element}
+ */
+function MenuLinkGroup({
+  menuItems,
+  href,
+  groupName,
+  activeIndex,
+  setActiveIndex,
+}) {
+  const router = useRouter();
+
+  return menuItems[groupName].map((item, index) => {
+    return (
+      <SubCategory
+        key={`subcategory-${item.title}`}
+        item={item}
+        index={index}
+        router={router}
+        href={href}
+        activeIndex={activeIndex}
+        setActiveIndex={setActiveIndex}
+      />
+    );
+  });
+}
+
+function SubCategory({ item, index, router, activeIndex, setActiveIndex }) {
```

Comment:
generelt brug es6 arrow funktioner `() => {}`

Reply:
synes ikke vi plejer at bruge dem i bib.dk?

Reply:
Det findes rundt omkring i bib. God vane at komme ind i, ikke at skrive til en forældet standard. Du har ikke brug for `this` i konteksten, så der er ingen grund til at bruge `function`

Reply:
Almindelige funktioner er ikke forældede. Det er bare forskellige måder at styre sit flow på.

Arrow-funktioner har (også) en del ulemper, så personligt synes jeg kun, at det er et yderst snæver use-case hvor den er mere praktisk end almindelige funktioner (hvis man ikke tænker f.eks. forwardRef i React som kun kan finde ud af at lave med arrow-funktioner).